### PR TITLE
analog: add commands

### PR DESCRIPTION
**Pull request by DeadByElpy**

### CODE REVIEW

File: src/analog/commands/downlink/getExAbsArchiveDaysMc.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+/**
+ * Encode command parameters.
+ *
+ * @param parameters - command payload
+ * @returns encoded bytes
+ */
+export const toBytes = ( parameters: IGetExAbsArchiveDaysMCParameters ): types.TBytes => {
+    const buffer: ICommandBinaryBuffer = new CommandBinaryBuffer(COMMAND_BODY_SIZE);
+    const {startTime2000, days, channelList} = parameters;
+    buffer.setDate(startTime2000);
```
Comment (DarkPark) [inline src/analog/commands/downlink/getExAbsArchiveDaysMc.ts:87]:
```suggestion

    buffer.setDate(startTime2000);
```

File: src/analog/commands/downlink/getExAbsArchiveHoursMc.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+ * [Command format documentation](https://github.com/jooby-dev/jooby-docs/blob/main/docs/analog/commands/GetExAbsArchiveHoursMC.md#request)
+ */
+
+import * as types from '../../../types.js';
+import * as command from '../../utils/command.js';
+import {TTime2000, getTime2000FromDate, getDateFromTime2000} from '../../utils/time.js';
+import CommandBinaryBuffer, {ICommandBinaryBuffer, IChannel} from '../../utils/CommandBinaryBuffer.js';
+
+
+interface IGetExAbsArchiveHoursMCParameters {
```
Comment (DarkPark) [inline src/analog/commands/downlink/getExAbsArchiveHoursMc.ts:28]:
```suggestion
interface IGetExAbsArchiveHoursMcParameters {
```
?

File: src/analog/commands/uplink/getExAbsArchiveDaysMc.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+interface IGetExAbsArchiveDaysMCResponseParameters {
+    channelList: Array<IChannelArchiveDaysAbsolute>;
+    startTime2000: TTime2000;
+    days: number;
+}
+
+
+export const id: types.TCommandId = 0x0d1f;
+export const name: types.TCommandName = 'getExAbsArchiveDaysMCResponse';
```
Comment (DarkPark) [inline src/analog/commands/uplink/getExAbsArchiveDaysMc.ts:47]:
```suggestion
export const name: types.TCommandName = 'getExAbsArchiveDaysMc';
```

File: src/analog/commands/downlink/getExAbsArchiveDaysMc.ts (hunk 1 above)
Comment (DeadByElpy) [reply src/analog/commands/downlink/getExAbsArchiveDaysMc.ts:87]: fixed

File: src/analog/commands/downlink/getExAbsArchiveHoursMc.ts (hunk 2 above)
Comment (DeadByElpy) [reply src/analog/commands/downlink/getExAbsArchiveHoursMc.ts]: fixed in 2 commands

File: src/analog/commands/downlink/getExAbsArchiveDaysMc.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+ * [Command format documentation](https://github.com/jooby-dev/jooby-docs/blob/main/docs/analog/commands/GetExAbsArchiveDaysMC.md#request)
+ */
+
+import * as types from '../../../types.js';
+import * as command from '../../utils/command.js';
+import {TTime2000, getTime2000FromDate} from '../../utils/time.js';
+import CommandBinaryBuffer, {ICommandBinaryBuffer, IChannel} from '../../utils/CommandBinaryBuffer.js';
+
+
+interface IGetExAbsArchiveDaysMCParameters {
```
[the remark this answers is not in the document]
Comment (DeadByElpy) [reply src/analog/commands/downlink/getExAbsArchiveDaysMc.ts]: fixed in all commands

File: src/analog/commands/uplink/getExAbsArchiveDaysMc.ts (hunk 3 above)
Comment (DeadByElpy) [reply src/analog/commands/uplink/getExAbsArchiveDaysMc.ts]: fixed

File: src/analog/commands/uplink/getExAbsArchiveHoursMc.ts [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+interface IGetArchiveHoursMCResponseParameters {
+    channelList: Array<IChannelHours>;
+    startTime2000: TTime2000;
+    hours: number;
+}
+
+
+export const id: types.TCommandId = 0x1a;
+export const name: types.TCommandName = 'getArchiveHoursMCResponse';
```
[the remark this answers is not in the document]
Comment (DeadByElpy) [reply src/analog/commands/uplink/getExAbsArchiveHoursMc.ts]: fixed

**Review by DarkPark — APPROVED** [review]

**Pull request merged**